### PR TITLE
Fixed get-html-tag to return a :tr inside a table

### LINKED ISSUE #18: Document tables and get-html-tag
When rendering widgets in tables, we can have a surprise of a badly rendered page. This is due to the fact of Weblocks surrounding its Widget html with a `div` which gets a proper id. But for tables, we don't want a `div`. The `get-html-tag` method of our widget must return `:tr`. Document this.

(I self-assign this.)

### PR DESCRIPTION
**Pull request by vindarel**
fixes #18 

Hi, first PR of code :)
I only tested manually.

Kuddos to Spinneret, really a good library.

### CODE REVIEW

**Comment by svetlyak40wt** [comment]
Great, however, I think there some cases when this code will produce a wrong result. For example, what if you are rendering a widget inside such structure:

```
<table>
  <tr>
    <td>{{ a widget }}</td>
  </tr>
</table>
```

or inside of
```
<table>
  <tr>
    {{ a widget }}
  </tr>
</table>
```

We need to look at the last tag in the path, not running a `find` on it.

And as an improvement, this code can return `:tr`, if parent tag is `:table` and `:td` if parent tag is  `:tr`.

What is interesting – there are no unit tests on widget's rendering. I'll create a pull request which add a test and then you can add your test nearby.

**Comment by svetlyak40wt** [comment]
Here is it: https://github.com/40ants/weblocks/pull/24 let's wait while tests on the Travis become green and I'll merge my pull. After that, please, rebase on `reblocks` branch and add few tests to check corner case for your improvement.

**Comment by vindarel** [comment]
update: on my new system, I can reproduce the CI's failure.

File: src/html.lisp [hunk 1]
```diff
@@ -29,7 +29,7 @@
   `(let ((spinneret:*html-lang* *lang*)
          (spinneret:*html* *stream*)
          ;; We want to an HTML which is nice to read, by default
-         (*print-pretty* t))
+         (*print-pretty* nil))  ;TODO: set as a parameter, easier for tests.
```
Comment (svetlyak40wt) [inline src/html.lisp]: Seems, you forgot to change it back.

File: t/widgets/render-methods.lisp [hunk 2]
```diff
@@ -32,11 +32,52 @@
...
+     (with-html
+       (:table
+        (:tr
+         (:td
+          (render widget)))))
+     ;; with (*print-pretty* nil) because indentation mixed of spaces and tab
+     ;; is difficult to get correctly.
+     ;; Even with it we need this weird indentation of 3 spaces.
+     "<table><tr><td><div class=\"widget widget-inside-table\"><p>foo bar</div></table>"
+     ;TODO: missing closing tags ??
```
Comment (svetlyak40wt) [inline t/widgets/render-methods.lisp]: HTML allows some tags to not have a closing part for example, for `<p>`. XHTML has more strict rules.

File: t/widgets/render-methods.lisp [hunk 3]
```diff
@@ -32,11 +32,52 @@
...
+
+(deftest test-widget-nested-into-table
+  (let ((widget (make-instance 'widget-inside-table)))
+    (is-html
+     (with-html
+       (:table
+        (:tr
+         (:td
+          (render widget)))))
+     ;; with (*print-pretty* nil) because indentation mixed of spaces and tab
```
Comment (svetlyak40wt) [inline t/widgets/render-methods.lisp]: Actually, `is-html` should be modified to ignore difference in newlines and indentation. Probably, the easiest way is to set `*print-pretty*` to `nil` there.

**Review by svetlyak40wt — CHANGES_REQUESTED** [review]
Few minor fixes, and it will be excellent!

**Comment by svetlyak40wt** [comment]
Also, I don't see where you are using `make`.

**Comment by svetlyak40wt** [comment]
And tests should be fixed. Seems they are broken because you've made `*print-pretty* = nil` by default. In case if you decide to modify `is-html` to set `*print-pretty*` to nil, these 3 failed tests should be fixed by changing the indentation of expected outputs.

**Comment by vindarel** [comment]
> Also, I don't see where you are using make.

oops, let's remove this Makefile, for this PR at least.

File: t/widgets/render-methods.lisp [hunk 4]
```diff
@@ -11,6 +11,7 @@
                 #:is-html))
 (in-package weblocks-test/widgets/render-methods)
 
+(setf weblocks/html:*pretty-html* nil)
```
Comment (svetlyak40wt) [inline t/widgets/render-methods.lisp]: Please, don't change this on the package level, because somebody can load `weblocks-test` system during their normal workflow. And the output will be changed not only for tests but also for the whole project.

Change `is-html` macro to bind `weblocks/html:*pretty-html*` to nil.

**Review by svetlyak40wt — CHANGES_REQUESTED** [review]

**Comment by vindarel** [comment]
I also get the remaining failing test. Need to have a closer look.

**Comment by svetlyak40wt** [comment]
Seems, this test should be fixed, because it checks html with formatting, like that:

```
<link rel=stylesheet type=text/css
           href=https://example.com/some.css
           media=screen>
```

you just need to remove newlines.

**Comment by vindarel** [comment]
oops, I know I tried though! Maybe I didn't recompile the test… *c'est la vie* :)

should be ok now. Thanks.

**Comment by svetlyak40wt** [comment]
Great job, @vindarel!
Pull is merged now!

**Comment by svetlyak40wt** [comment]
Thank you!

**Comment by vindarel** [comment]
and thank you !
